pkg/bootstrap/option/convert.go

```diff
@@ -180,6 +181,32 @@ func durationConverter(value *types.Duration) convertFunc {
 	}
 }
 
+// openCensusAgentContextConverter returns a converter that returns the list of
+// distributed trace contexts to propagate with envoy.
+func openCensusAgentContextConverter(contexts []meshAPI.Tracing_OpenCensusAgent_TraceContext) convertFunc {
+	return func(*instance) (interface{}, error) {
+		if len(contexts) == 0 {
+			return convertToJSON([]string{"TRACE_CONTEXT", "GRPC_BIN", "CLOUD_TRACE_CONTEXT", "B3"}), nil
+		}
+
+		var envoyContexts []string
+		for _, c := range contexts {
+			switch c {
+			case meshAPI.Tracing_OpenCensusAgent_W3C_TRACE_CONTEXT:
+				envoyContexts = append(envoyContexts, "TRACE_CONTEXT")
+			case meshAPI.Tracing_OpenCensusAgent_GRPC_BIN:
+				envoyContexts = append(envoyContexts, "GRPC_BIN")
+			case meshAPI.Tracing_OpenCensusAgent_CLOUD_TRACE_CONTEXT:
+				envoyContexts = append(envoyContexts, "CLOUD_TRACE_CONTEXT")
+			case meshAPI.Tracing_OpenCensusAgent_B3:
+				envoyContexts = append(envoyContexts, "B3")
+				// Ignore UNSPECIFIED
```

Comment:
super minor nit: i might move this comment up to the top above W3_TRACE_CONTEXT, as the indent makes it a bit funky at the bottom.

Reply:
thanks, that looks better now.